src/com/facebook/buck/android/dalvik/DalvikStatsCache.java

```diff
@@ -33,7 +33,7 @@
 
   DalvikStatsTool.Stats getStats(FileLike entry) {
     String name = entry.getRelativePath();
-    if (!name.endsWith(".class")) {
+    if (!name.endsWith(".class") || name.endswith("module-info.class")) {
```

Comment:
Please fix the typo "endswith"

Reply:
Similar issue is https://stackoverflow.com/questions/45311295/error-scanning-entry-module-info-class-when-starting-jetty-server.
Should upgrading ClassReader.class work? we use ClassReader to classes and it can't read the new format.

Reply:
Ignoring this class should work. Even if we read it, it would be a waste as we don't need to use any of this info for estimating dex size.

Reply:
@kageiit makes sense.

Reply:
Just don't want to add to many special cases. If we do, @shubhraprakash please comment why.

Reply:
Speaking of special cases, we would have to rework this pipeline a bit to handle multi release jars eventually as duplicate class files will be inside META-INF for different versions of the jvm. Thats a separate issue however

Reply:
@v-jizhang  who is  @shubhraprakash, FYI this is @surapuramakhil. And added comment for special case

Reply:
Sorry :)